module2/projects/little_shop/requirements.md

```diff
@@ -0,0 +1,659 @@
+
+#### DRY Exception Handlers in Code
+
+As part of the project, we're requiring students try to refactor exception handling and sad path logic in their controllers as much as possible. Take a look at [this page](./dry_exception_handling) to learn more about some options. 
+
+#### Sad Path vs Edge Case
+
+Sad Path: the user did something which didn't cause an _error_ but didn't work out the way they'd hoped. For example, searching for a merchant by name and getting zero results is a "sad path"
+
+Edge Case: the user did something which broke the functionality of an endpoint. For example, a user searches for an item based on a negative price, or searching between revenue dates where the end date comes before the start date.
+
+## SECTION ONE: RESTful Endpoints, Minimum Requirements:
+
+You will need to expose the following RESTful API endpoints for the following:
+
+* Merchants:
+  * get all merchants
+    * get all merchants sorted by newest to oldest
+    * get all merchants with returned items (check invoice)
+    * get all merchants with calculated count of items
+  * get one merchant
+  * create a merchant
+  * edit a merchant
+  * delete a merchant
+  * get all items for a given merchant ID
+
+* Items:
+  * get all items
+  *   get all items sorted by price (low to high)
+  * get one item
+  * create an item
+  * edit an item
+  * delete an item
+  * get the merchant data for a given item ID
+
+* Additional RESTful endpoints
+  * get all customer names for a given merchant
+  * get all invoices for a given merchant, filtered by status
+
+
+## SECTION TWO: Non-RESTful Search Endpoints
+
+You will get to choose ONE group of endpoint pairs from the following list:
+
+**A. One Merchant & All Items:**
+  * Find one MERCHANT based on search criteria, and
+  * Find all ITEMS based on search criteria
+
+Or, 
+
+**B. One Item & All Merchants:**
+  * Find one ITEM based on search criteria, and
+  * Find all MERCHANTS based on search criteria
+
+## Your Project MVP
+
+In total, the MINIMUM requirement will be 16 endpoints:
+
+* section one has 14 endpoints (some with additional features via query parameters)
+* section two has 2 endpoints
+
+
+You may choose to divide these up between your project partners in whatever way seems best. One pair could handle the Merchant stories, and one pair could handle the Item and additional endpoints. You may also decide to work on the first story/stories together to have a solid understanding first before dividing and conquering. 
+
+You can reference the [Wireframes](./wireframes) to get a better idea of how these endpoints might be used in a frontend application.
+
+---
+
+# 3. API requests/responses, more detail
+# SECTION ONE: RESTful requests
+
+<details><summary><h3>1. Fetch all Items/Merchants</h3></summary>
```

Comment:
Are you open to using the dropdown formatting we've used in lessons for these dropdowns? I think it makes it easier to tell at a glance that its a drop down and will keep things more consistent?

The syntax is easier than how you have to create a dropdown in regular markdown.

```
<section class="dropdown">
### 1. Fetch all Posters

These "index" endpoints should:
....
</section>
```

Reply:
![image](https://github.com/user-attachments/assets/976e61c2-6538-453f-98a5-ba80f8c76745)
 
It would format like this

Reply:
Updating all of these!